lib/syskit/roby_app/plugin.rb

```diff
@@ -158,51 +152,60 @@ def self.setup(app)
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
 
                 # Log Transfer FTP Server spawned during Application#setup
-                setup_local_log_transfer_server(app) if log_transfer_ip
+                app.setup_local_log_transfer_server if app.log_transfer_ip
             end
 
-            def self.setup_local_log_transfer_server(app)
+            def setup_local_log_transfer_server
                 @tmp_root_ca = TmpRootCA.new(log_transfer_ip)
-                start_local_log_transfer_server(app.log_dir, @tmp_root_ca)
-                config_log_transfer
+                @log_transfer_password = SecureRandom.base64(15)
+                start_local_log_transfer_server(log_dir, @tmp_root_ca)
             end
 
-            def self.send_file_transfer_command(name, logfile)
+            def send_file_transfer_command(name, logfile)
                 # Establishes communication with said process server
                 client = Syskit.conf.process_server_for(name)
 
                 # Commands method log_upload_file from said process server
                 client.log_upload_file(
-                    @log_transfer_ip,
-                    @log_transfer_port,
-                    @log_transfer_certificate,
-                    @log_transfer_user,
+                    log_transfer_ip,
+                    log_transfer_port,
+                    log_transfer_certificate,
+                    log_transfer_user,
```

Comment:
Seems to me that you don't actually need the attributes anymore. They could be private to the Plugin module (apart from log_transfer_ip). Is that right ?

Reply:
That is right. I think we never really did, actually.